### PR TITLE
Use new VQE workflow if PennyLane version > 0.17

### PR DESCRIPTION
**Pull request by mariaschuld**
Since we now want to advertise the faster VQE workflow, we better use it in the benchmarks.

I also deleted that "delete-me" folder :)

I know we don't have tests, but I checked locally that the if statement is used correctly depending on the version.

### CODE REVIEW

File: benchmarks/benchmark_functions/default_settings.py [hunk 1]
```diff
@@ -24,8 +24,6 @@
 from pennylane.templates import BasicEntanglerLayers
 from pennylane.templates.decorator import template as template_decorator
 from pennylane.templates.subroutines import UCCSD
-from pennylane import Identity, PauliX, PauliY, PauliZ
-
```
Comment (mariaschuld) [inline benchmarks/benchmark_functions/default_settings.py:28]: Was obsolete before

File: benchmarks/benchmark_functions/default_settings.py [hunk 2]
```diff
@@ -87,15 +85,13 @@ def _vqe_defaults(hyperparams):
     device = hyperparams.pop("device", "default.qubit")
     interface = hyperparams.pop("interface", "autograd")
     diff_method = hyperparams.pop("diff_method", "best")
-    optimize = hyperparams.pop("optimize", True)
-
-    options_dict = {"interface": interface, "diff_method": diff_method, "optimize": optimize}
+    grouping = hyperparams.pop("optimize", False)
```
Comment (soranjh) [inline benchmarks/benchmark_functions/default_settings.py]: Should we benchmark with the grouping option turned on? In general, users use the faster method.

File: benchmarks/benchmark_functions/vqe.py [hunk 3]
```diff
@@ -42,11 +43,21 @@ def benchmark_vqe(hyperparams={}):
                     * 'optimize': argument for grouping the observables composing the Hamiltonian
     """
 
-    ham, ansatz, params, n_steps, device, options_dict = _vqe_defaults(hyperparams)
+    ham, ansatz, params, n_steps, device, interface, diff_method, grouping = _vqe_defaults(hyperparams)
 
-    cost_fn = qml.ExpvalCost(ansatz, ham, device, **options_dict)
+    if version.parse(qml.__version__) > version.parse("0.17"):
```
Comment (soranjh) [inline benchmarks/benchmark_functions/vqe.py:48]: I am curious how faster the new workflow is?

**Review by soranjh — APPROVED** [review]
Thanks Maria, just approved.

File: benchmarks/benchmark_functions/default_settings.py (hunk 2 above)
Comment (mariaschuld) [reply benchmarks/benchmark_functions/default_settings.py]: Sounds good!

Comment (mariaschuld) [inline benchmarks/benchmark_functions/default_settings.py]:
```suggestion
    grouping = hyperparams.pop("optimize", True)
```

File: benchmarks/benchmark_functions/vqe.py (hunk 3 above)
Comment (mariaschuld) [reply benchmarks/benchmark_functions/vqe.py:48]: Unfortunately only 0.25x as fast. Previously, when we tried to implement our own sparse method it was about 0.07x as fast. We still want to get there!

**Review by albi3ro — APPROVED** [review]
No comments from me.  

Looking at how many things are returned from `_vqe_defaults` we map want to return things in the form of a dictionary in the future, but that's a thought for a later date.

**Pull request merged**